server/sonar-server/src/main/java/org/sonar/server/component/ws/ProvisionedProjectsAction.java

```diff
@@ -0,0 +1,129 @@
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.component.ws;
+
+import com.google.common.io.Resources;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.server.ws.WebService.Param;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.es.SearchOptions;
+import org.sonar.server.user.UserSession;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+public class ProvisionedProjectsAction implements ProjectsAction {
+  private final DbClient dbClient;
+
+  private static final String[] POSSIBLE_FIELDS = new String[] {"uuid", "key", "name", "creationDate"};
+
+  public ProvisionedProjectsAction(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller
+      .createAction("provisioned")
+      .setDescription(
+        "Get the list of provisioned projects.<br /> " +
+          "Require admin role.")
+      .setSince("5.2")
+      .setResponseExample(Resources.getResource(getClass(), "projects-example-provisioned.json"))
+      .setHandler(this)
+      .addPagingParams(100)
+      .addFieldsParam(Arrays.asList(POSSIBLE_FIELDS));
+
+    action
+      .createParam(Param.TEXT_QUERY)
+      .setDescription("UTF-8 search query")
+      .setExampleValue("sonar");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkGlobalPermission(UserRole.ADMIN, "You need admin rights.");
+    SearchOptions options = new SearchOptions().setPage(
+      request.mandatoryParamAsInt(Param.PAGE),
+      request.mandatoryParamAsInt(Param.PAGE_SIZE)
+      );
+    List<String> desiredFields = desiredFields(request);
+    String query = request.param(Param.TEXT_QUERY);
+
+    DbSession dbSession = dbClient.openSession(false);
+    List<ComponentDto> projects;
```

Comment:
+1 for saving lines and being less ambiguous. I'm not sure it would apply if the computation after finally was costly. Anyway, I apply your feedback.